dom.go

```diff
@@ -650,14 +654,7 @@ func (l keyedList) reconcile(parent *HTML, prevChild ComponentOrHTML) (pendingMo
 
 // remove keyedList elements from the parent.
 func (l keyedList) remove(parent *HTML) {
-	// Copy parent scope to remove list elements from
-	l.html.node = parent.node
-	l.html.insertBeforeNode = parent.insertBeforeNode
-	l.html.removeChildren(l.html.children)
-	// Update parent insertBeforeNode in case any children were removed.
-	if parent.insertBeforeNode != nil {
-		parent.insertBeforeNode = l.html.insertBeforeNode
-	}
+	parent.removeChildren(l.html.children)
```

Comment:
From what I gather this is a much simpler implementation and does the same exact thing because state inside `l.html` is never used once this function is called.

Maybe disagree, though?

Reply:
The insertion node will be advanced to the next sibling if it matches any of the removed children (in `*HTML.removeChild()`), so the code here needs to stay as is.

Reply:
But only `parent.insertBeforeNode` matters after this point. And `parent.removeChildren` handles updating this. So I don't see where `l.html.insertBeforeNode` is used once this function returns?

Reply:
Ah, I see.  The insertion node updates here are to handle nested list rendering - we may remove nodes down in a nested list, and need to bubble up changes to the insertion node.

This change does make for less lines, but I'm concerned about the possibility of introducing bugs if the logic in `removeChildren` changes.  The code as it stands maintains a controlled boundary between the parent and list.

Reply:
Ok, in that case I will revert + update docs slightly. I can understand the argument that this is better in terms of correctness.

Reply:
(done)